cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -610,6 +623,12 @@ func (c *machineController) findScalableResourceProviderIDs(scalableResource *un
 
 		if found {
 			if providerID != "" {
+				// If the machine is deleting, prepend the deletion guard on the provider id
+				// so that it can be properly filtered when counting the number of nodes and instances.
+				if !machine.GetDeletionTimestamp().IsZero() {
+					klog.V(4).Infof("Machine %q has a non-zero deletion timestamp", machine.GetName())
+					providerID = fmt.Sprintf("%s%s_%s", deletingMachinePrefix, machine.GetNamespace(), machine.GetName())
```

Comment:
let's hoist this string transformation into a dedicated func w/ UT

Reply:
added